libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/telemetryevents/TelemetryStep.kt

```diff
@@ -0,0 +1,43 @@
+package com.mapbox.navigation.core.telemetry.telemetryevents
+
+import com.mapbox.navigation.base.trip.model.RouteLegProgress
+
+/**
+ * Documentation is here [https://paper.dropbox.com/doc/Navigation-Telemetry-Events-V1--AuUz~~~rEVK7iNB3dQ4_tF97Ag-iid3ZImnt4dsW7Z6zC3Lc]
+ */
+
+// Defaulted values are optional
+
+data class TelemetryStep(
+    val upcomingInstruction: String,
+    val upcomingType: String,
+    val upcomingModifier: String,
+    val upcomingName: String,
+    val previousInstruction: String,
+    val previousType: String,
+    val previousModifier: String,
+    val previousName: String,
+    val distance: Int,
+    val duration: Int,
+    val distanceRemaining: Int,
+    val durationRemaining: Int
+)
+
+fun TelemetryStep.populateTelemetryStep(legProgress: RouteLegProgress): TelemetryStep = apply {
```

Comment:
This is duplicated in [`TelemetryUtils`](https://github.com/mapbox/mapbox-navigation-android/blob/60ab4bce65046ed735a0dca9a6efaebf9deae4ec/libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/TelemetryUtils.kt) and never used.

Reply:
👍